index-providers/modeshape-elasticsearch-index-provider/src/main/java/org/modeshape/jcr/index/elasticsearch/SearchResults.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * ModeShape (http://www.modeshape.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.modeshape.jcr.index.elasticsearch;
+
+import org.elasticsearch.action.search.SearchRequestBuilder;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.modeshape.jcr.cache.NodeKey;
+import org.modeshape.jcr.spi.index.Index;
+import org.modeshape.jcr.spi.index.ResultWriter;
+
+/**
+ *
+ * @author kulikov
+ */
+public class SearchResults implements Index.Results {
+
+    private final SearchRequestBuilder searchBuilder;
+    private SearchResponse response;
+    private int pos = 0;
+
+    /**
+     * Creates new search result instance for the given search request.
+     * 
+     * @param searchBuilder search request as request builder.
+     */
+    public SearchResults(SearchRequestBuilder searchBuilder) {
+        this.searchBuilder = searchBuilder;
+        this.response = null;
+    }
+
+    @Override
+    public boolean getNextBatch(ResultWriter writer, int batchSize) {
+        try {
+            response = searchBuilder
+                    .setSize(batchSize)
+                    .setFrom(pos)
+                    .execute()
+                    .actionGet();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Is there no better way to handle this exception? At least it should be logged and not printed.